storage/ftp_storage.py

```diff
@@ -139,13 +139,19 @@ def save_to_file(self, out_file: BinaryIO) -> None:
         ftp_client = self._connect()
         filename = self._cd_to_file(ftp_client)
 
-        ftp_client.retrbinary("RETR {0}".format(filename), callback=out_file.write)
+        try:
+            ftp_client.retrbinary("RETR {0}".format(filename), callback=out_file.write)
+        except error_perm:
```

Comment:
Similar to the S3 implementation's `ClientError`, the `error_perm` class is a generic exception for the set of FTP 5xx error codes. The FTP error code for "file unavailable" is 550 (see [RFC 959](https://tools.ietf.org/html/rfc959) section 4.2.1).

Also, we should chain the original exception.